### PR TITLE
Apply storybook for example first component

### PR DESCRIPTION
**Pull request by thanhtrantruong47**
Apply storybook for example first component

https://storybook.js.org/docs/get-started

image

### CODE REVIEW

File: examples/first-component/chromatic.config.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (hungdo-agilityio) [inline examples/first-component/chromatic.config.json:1]: I just saw this fille was list in `.gitignore`, please check

File: examples/first-component/package.json [hunk 2]
```diff
@@ -7,22 +7,37 @@
...
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "storybook": "storybook dev -p 6006",
+    "build-storybook": "storybook build",
+    "chromatic": "npx chromatic --project-token=chpt_b1ff1859b6256dd"
   },
   "dependencies": {
-    "react": "^18.2.0",
+    "react": "^18.3.1",
```
Comment (hungdo-agilityio) [inline examples/first-component/package.json]: Why react bump up version

File: examples/first-component/src/types/Person.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export interface Person {
-  id: string;
+  id?: string;
```
Comment (hungdo-agilityio) [inline examples/first-component/src/types/Person.ts]: Why modify id to optional

File: examples/first-component/.gitignore [hunk 4]
```diff
@@ -22,3 +22,7 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
+
+*storybook.log
+storybook-static
+chromatic.config.json
```
Comment (hungdo-agilityio) [inline examples/first-component/.gitignore]: Why we list `chromatic.config.json` into `.gitignore`

File: examples/first-component/src/components/ScientistList/ScientistsList.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import type { Meta, StoryObj } from "@storybook/react";
+
+import ScientistList from "./index";
+const meta: Meta<typeof ScientistList> = {
+  component: ScientistList,
+};
+
+export default meta;
+type Story = StoryObj<typeof ScientistList>;
+const scientist = [
```
Comment (thanhnguyen-agilityio) [inline examples/first-component/src/components/ScientistList/ScientistsList.stories.tsx]: `scientist` is a single noun. A list should have naming plural.

File: examples/first-component/chromatic.config.json (hunk 1 above)
Comment (thanhtrantruong47) [reply examples/first-component/chromatic.config.json:1]: Fixed

File: examples/first-component/src/types/Person.ts (hunk 3 above)
Comment (thanhtrantruong47) [reply examples/first-component/src/types/Person.ts]: Fixed

File: examples/first-component/src/components/ScientistList/ScientistsList.stories.tsx (hunk 5 above)
Comment (thanhtrantruong47) [reply examples/first-component/src/components/ScientistList/ScientistsList.stories.tsx]: Fixed

File: examples/first-component/.gitignore (hunk 4 above)
Comment (thanhtrantruong47) [reply examples/first-component/.gitignore]: Fixed

File: examples/first-component/package.json (hunk 2 above)
Comment (thanhtrantruong47) [reply examples/first-component/package.json]: Fixed

**Review by thanhnguyen-agilityio — APPROVED** [review]

**Pull request merged**